eppo_core/src/events/vec_event_queue.rs

```diff
@@ -0,0 +1,197 @@
+use crate::events::queued_event::{QueuedEvent, QueuedEventStatus};
+use std::collections::VecDeque;
+use std::sync::{Arc, Mutex};
+
+pub trait EventQueue {
+    /// Pushes an event to the end of the queue.
+    fn push(&self, event: QueuedEvent) -> Result<(), QueueError>;
+
+    /// Returns the next batch of events with the provided status.
+    /// and with status PENDING. Events are returned in FIFO order.
+    fn next_batch(&self, status: QueuedEventStatus) -> Vec<QueuedEvent>;
+
+    /// Returns whether the queue contains enough events for delivering *at least* one batch.
+    fn is_batch_full(&self) -> bool;
+}
+
+/// A simple event queue that stores events in a vector
+#[derive(Debug, Clone)]
+pub struct VecEventQueue {
+    batch_size: usize,
+    max_queue_size: usize,
+    event_queue: Arc<Mutex<VecDeque<QueuedEvent>>>,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum QueueError {
+    QueueFull,
+}
+
+// batch size of zero means each event will be delivered individually, thus effectively disabling batching.
+const MIN_BATCH_SIZE: usize = 0;
+const MAX_BATCH_SIZE: usize = 10_000;
+
+impl VecEventQueue {
```

Comment:
Basically brought back the initial implementation from #132 and expanded upon it to add max_queue_size and filtering by event status